src/dawn/CodeGen/GridTools/GTCodeGen.cpp

```diff
@@ -821,19 +905,65 @@ std::string GTCodeGen::generateStencilInstantiation(
                        });
   }
 
-  ASTStencilDesc stencilDescCGVisitor(stencilInstantiation.get(), stencilIDToStencilNameMap,
+  StencilWrapperConstructor.commit();
+
+
+  // Generate the run method by generate code for the stencil description AST
```

Comment:
this method is quite large already. At some point it would make sense to organize it with calls to 
``` generate_run(stencil) ```
```generate_ctr(stencil) ```

to make it more readable